src/swsssdk/dbconnector.py

```diff
@@ -27,90 +67,134 @@ def load_sonic_db_config(sonic_db_file_path=SONIC_DB_CONFIG_FILE):
+        return SonicDBConfig.get_instance(db_name, namespace)["unix_socket_path"]
 
     @staticmethod
-    def get_port(db_name):
+    def get_hostname(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig.get_instance(db_name)["port"]
+        return SonicDBConfig.get_instance(db_name, namespace)["hostname"]
 
     @staticmethod
-    def get_dbid(db_name):
+    def get_port(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig._sonic_db_config["DATABASES"][db_name]["id"]
+        return SonicDBConfig.get_instance(db_name, namespace)["port"]
 
     @staticmethod
-    def get_separator(db_name):
+    def get_dbid(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig._sonic_db_config["DATABASES"][db_name]["separator"]
+        SonicDBConfig.db_name_validation(db_name, namespace)
+        return SonicDBConfig._sonic_db_config[namespace]["DATABASES"][db_name]["id"]
+
+    @staticmethod
+    def get_separator(db_name, namespace=''):
+        if SonicDBConfig._sonic_db_config_init == False:
+            SonicDBConfig.load_sonic_db_config()
+        SonicDBConfig.db_name_validation(db_name, namespace)
+        return SonicDBConfig._sonic_db_config[namespace]["DATABASES"][db_name]["separator"]
 
 class SonicV2Connector(DBInterface):
-    def __init__(self, use_unix_socket_path=False, **kwargs):
+    def __init__(self, use_unix_socket_path=True, **kwargs):
         super(SonicV2Connector, self).__init__(**kwargs)
         self.use_unix_socket_path = use_unix_socket_path
+        self.namespace = ''
+
+        # Load sonic DB configs
+        self.load_db_configs()
+
         for db_name in self.get_db_list():
             # set a database name as a constant value attribute.
             setattr(self, db_name, db_name)
 
-    def connect(self, db_name, retry_on=True):
+    def connect(self, db_name, retry_on=True, namespace=''):
+        if namespace == None:
+            msg = "{} is not a valid namespace name".format(namespace)
+            logger.warning(msg)
+            raise RuntimeError(msg)
+
+        # The TCP connection to a DB in different namespace in not supported.
+        if namespace != '' and self.use_unix_socket_path == False:
+            message = "TCP connectivity to the DB instance in a different namespace is not implemented!"
+            raise NotImplementedError(message)
+
+        self.namespace = namespace
+        for dbname in self.get_db_list():
+            # set a database name as a constant value attribute.
+            setattr(self, dbname, dbname)
```

Comment:
Why add this part? It is also inefficient to execute for each `connect()`

Reply:
I have to do it per connect, where the user specifies the namespace and the databases could be different as defined in the database_config.json ( present in that namespace ). I think it is better to assume that the dbnames could be different in different namespaces ?

I could probably add a check to see if the namespace user given is the same as default namespace '' -- in which case I will ignore doing setattr again ( as for default namespace '' it is already done in __init__ constructor of SonicV2Connector class )

Reply:
I agree the dbnames could be different in different namespaces. What is the use case of these constant value attributes? I cannot find one in this PR.

---
In reply to: [402695930](https://github.com/Azure/sonic-py-swsssdk/pull/63#discussion_r402695930) [](ancestors = 402695930)